### PR TITLE
[native_assets_builder] Document rolling constraints

### PR DESCRIPTION
**Pull request by dcharkes**

### CODE REVIEW

File: pkgs/native_assets_builder/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Dartdev and Flutter Tools are the commandline interfaces for `dart ...` and `flutter ...` commands.
+
+## Downstream builds
+
+Both Dartdev and Flutter Tools are built in open source, and inside g3.
+
+* Dartdev as part of the Dart SDK build on the [Dart CI].
+* Dartdev as part of the g3 build (see CBuild comments on CLs on [Gerrit]).
+* Flutter Tools as part of the workflows on GitHub.
+* Flutter Tools as part of the g3 build (see the "Google testing" workflow on PRs on GitHub).
```
Comment (mosuem) [inline pkgs/native_assets_builder/CONTRIBUTING.md:12]: Add a link here and in the other places where they are not present yet.

File: pkgs/native_assets_builder/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+The versions used in these different places are as follows:
+
+* The Dart SDK uses pinned dependencies in [DEPS], the current has is in `native_rev`.
```
Comment (mosuem) [inline pkgs/native_assets_builder/CONTRIBUTING.md:16]:
```suggestion
* The Dart SDK uses pinned dependencies in [DEPS], the current hash is in `native_rev`.
```

File: pkgs/native_assets_builder/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+The above means the following.
+
+1. The DEPS in the Dart SDK can only be rolled forward as much as is still compatible with the published deps currently used in Flutter Tools.
+2. Flutter Tools can only be rolled forward as much as is still compatible with with the deps pinned in the Dart SDK.
+
+So, any breaking change must be done in the following way:
+
+1. Introduce a new API.
+2. Release a published version.
```
Comment (mosuem) [inline pkgs/native_assets_builder/CONTRIBUTING.md:30]:
```suggestion
2. Release a new major version.
```

**Review by mosuem — APPROVED** [review]

File: pkgs/native_assets_builder/CONTRIBUTING.md (hunk 3 above)
Comment (dcharkes) [reply pkgs/native_assets_builder/CONTRIBUTING.md:30]: I think we only need to bump the major version when removing the old API right?

File: pkgs/native_assets_builder/CONTRIBUTING.md (hunk 1 above)
Comment (dcharkes) [reply pkgs/native_assets_builder/CONTRIBUTING.md:12]: I don't know where to link to. I just know it shows up in the GitHub UI on your PR.

**Pull request merged**